Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -664,16 +669,221 @@ namespace xr
+                        status = ArAugmentedImageDatabase_addImage(
+                                xrContext->Session,
+                                augmentedImageDatabase,
+                                "",
+                                grayscale_buffer,
+                                image.width,
+                                image.height,
+                                image.width,
+                                &index);
+                    }
+
+                    if (status == AR_SUCCESS)
+                    {
+                        trackableImagesCount++;
+                        scores.push_back(System::Session::Frame::ImageTrackingScore::TRACKABLE);
+                    }
+                    else
+                    {
+                        scores.push_back(System::Session::Frame::ImageTrackingScore::UNTRACKABLE);
+                    }
+                }
+                catch (std::exception)
+                {
+                    delete grayscale_buffer;
+                    throw;
+                }
+
+                delete grayscale_buffer;
+            }
+
+            // If we had at least one trackable image, set up image tracking.
+            if (trackableImagesCount > 0)
+            {
+                // Create an ArConfig
+                ArConfig *arConfig{};
+                ArConfig_create(xrContext->Session, &arConfig);
+                ArSession_getConfig(xrContext->Session, arConfig);
+
+                // Configure the ArSession to include the image tracking database
+                ArConfig_setAugmentedImageDatabase(xrContext->Session, arConfig,
+                                                   augmentedImageDatabase);
+                const ArStatus status = ArSession_configure(xrContext->Session, arConfig);
+
+                // If we failed to configure the session, error out.
+                if (status != AR_SUCCESS)
+                {
+                    throw std::runtime_error{"Failed to configure AR Session for Image Tracking"};
+                }
+
+                // Clean up the ArConfig.
+                ArConfig_destroy(arConfig);
+            }
+
+            return scores;
+        }
+
+        void UpdateImageTrackingResults(std::vector<Frame::ImageTrackingResult::Identifier>& updatedResults)
+        {
+            // Get list of updated images in the current frame.
+            int32_t imageListSize;
+            ArFrame_getUpdatedTrackables(xrContext->Session, xrContext->Frame, AR_TRACKABLE_AUGMENTED_IMAGE, trackableList);
+            ArTrackableList_getSize(xrContext->Session, trackableList, &imageListSize);
+
+            // For each updated image, get the current status.
+            for (int i = 0; i < imageListSize; ++i) {
+                ArTrackable* trackable = nullptr;
+                ArTrackableList_acquireItem(xrContext->Session, trackableList, i, &trackable);
+                ArAugmentedImage* imageTrackable = ArAsAugmentedImage(trackable);
```

Comment:
Uniform initializers for consistency?
```suggestion
                ArTrackable* trackable{nullptr};
                ArTrackableList_acquireItem(xrContext->Session, trackableList, i, &trackable);
                ArAugmentedImage* imageTrackable{ArAsAugmentedImage(trackable)};
```